### PR TITLE
7626 fix unknown currency error

### PR DESCRIPTION
**Pull request by shen-sat**
#### What? Why?

Closes #[the issue number this PR is related to] https://github.com/openfoodfoundation/openfoodnetwork/issues/7626

New developers have noticed the following error when running the application/tests/console for the _second_ time. 
```
rake aborted!
Money::Currency::UnknownCurrency: Unknown currency '
```
This error is always produced if they do not run script/setup (and therefore do not have a config/application.yml file) before running application/tests/console for the _first_ time.

The sequence of events is as follows (I’ve verified these steps using byebug):

1. clone repo
2. dev runs `bundle exec [any command]`
3. The line `preference :currency, :string, default: "USD"` is hit in `config/application.rb`
4. `Money.default_currency = Money::Currency.new(Spree::Config[:currency])` is hit in `config/initializers/money.rb` - Spree::Config[:currency] is not set at this point and so it defaults to the default value set in the previous step (ie it is set to “USD”)
5. `Spree::Config['currency'] = ENV['CURRENCY']` is hit in `config/application.rb` - at this point, because we don’t have a config/application.yml file, ENV[‘CURRENCY’] does not exist and this sets Spree::config[:currency] to an empty string. **This empty string is cached**
6. dev runs `bundle exec [any command]` (second time)
7. `preference :currency, :string, default: "USD"` is hit in `config/application.rb`
8. `Money.default_currency = Money::Currency.new(Spree::Config[:currency])` is hit in `config/initializers/money.rb` - this time Spree::Config[:currency] is set to the cached value (an empty string), and this leads to the error.

Please note: This error is not solved by running `script/setup` (ie creating the config/application.yml) file, because even with ENV vars now present the following steps occur 
1. `script/setup` (ie config/application.yml is created and ENV vars are now available)
2. `bundle exec [any command]` (a third time)
3. `preference :currency, :string, default: "USD"` is hit in `config/application.rb`
4. `Money.default_currency = Money::Currency.new(Spree::Config[:currency])` is hit in `config/initializers/money.rb` -  Spree::Config[:currency] is still set to the cached value (an empty string), and this again leads to the same error.
5. And so on…in a loop

As you can see, the problem is that even with ENV vars available, because the application never gets to set  `Spree::Config['currency'] = ENV['CURRENCY']`, Spree::Config['currency'] remains set to an empty string and the application never loads past this line.

This PR therefore does two things:
- Strongly encourages new devs to run `script/setup` before doing anything else
- Replaces  `Money.default_currency = Money::Currency.new(Spree::Config[:currency])` with `Money.default_currency = Money::Currency.new(ENV.fetch(‘CURRENCY’)) `. This will still raise an error at the same line if ENV vars are not present, but this time:
        - the error will be more descriptive (`key not found: "CURRENCY" (KeyError)`)
        - once ENV[‘CURRENCY’] has been set, the application will no longer raise an error
        - we will no longer depend on a cached Spree::Config[:currency] when booting the application, which is safer as it avoids the potential for the error loop described above

#### What should we test?
1. git clone the repo
2. try running any `bundle exec` command twice (you should see an error)
3. run `script/setup`
4. try running any `bundle exec` command (you should no longer see an error)

#### Release notes

I'm not sure what to write here. Maybe: fix unknown currency error for new developers on setup 

Changelog Category: Technical changes

#### Dependencies

No

#### Documentation updates

I've already changed the relevant pages

### CODE REVIEW

File: db/default/users.rb
Comment (shen-sat) [inline db/default/users.rb:65]: I changed this because when running `script/setup` this line is misleading (when it prints "Done!" to console it looks as if the entire setup process is complete when in fact it is not)

**Comment by andrewpbrett** [comment]
Sorry this has been sitting for a while! I just pressed the button to run the tests and moved it to the Code Review column on our board.

**Comment by shen-sat** [comment]
No worries @andrewpbrett, thanks for the update 🙂
Once/if this PR is merged, epic https://github.com/openfoodfoundation/openfoodnetwork/issues/7613 will almost be done 💪

**Review by mkllnk — APPROVED** [review]
Nice! :heart_eyes: 

This looks great!

**Review by mkllnk — APPROVED** [review]
I did the dev-test. Thank you for the detailed explanation. Since I had a working setup already, I needed to reset my database to re-produce the issue. I can confirm that your change fixes it. :heavy_check_mark: 

So this can be merged after the next approval.

**Pull request merged**
